### ISSUE TITLE
NerdFont glyphs get truncated. 

### ISSUE THREAD

**Issue by MichielVanderlee** [mintty/mintty#680]
Using wsltty 1.7.9 with mintty 2.7.9
This seems to be related to #638 but I seem to be missing something.

I've tried all [SauceCodePro regular fonts](https://github.com/ryanoasis/nerd-fonts/tree/master/patched-fonts/SourceCodePro/Regular/complete) All show up as small icons except for 'Sauce Code Pro Nerd Font Complete.ttf' which shows up as truncated, but the size I want.

I've tried tried executing:
```bash
$ cat
^[]77119;1^G   # Ctrl+[, ]77119;1, Ctrl+g

$ cat
^[]77119;2^G
```
but neither worked.

**Comment by mintty** [comment]
Which characters do you try to display? OSC 77119 only affects specific ranges of Unicode characters, e.g. Indic, not any iconic or even private use characters.
And what makes your case distinct from 638?

**Comment by DDuTCH** [comment]
Characters in the private space. 
F113 for example.
And it looks like 638 was solved, I can't figure out how though.

I've also are playing with the source code and compiling my own version of mintty, with no success so far. I've added the private space to wide.t with no effect. 
I've ensured all private space characters get the attr ATTR_WIDE, but it's still clipped on the left side, and the first character after is hidden under the right side of the icon.

**Comment by mintty** [comment]
638 was mostly about overhanging glyphs, for which support was added and released.
Note that you cannot really expect particular rendering for private use characters.
It would help though to have a precise symptom description, with exact output and screenshot.

**Comment by DDuTCH** [comment]
Original:
![image](https://user-images.githubusercontent.com/918128/29501524-65854b7e-85ff-11e7-8c05-1192cc5b2d1e.png)

After making all private space characters wide. 
term.c line:1179
```
if (tchar >= 0xE000 && tchar < 0xF900) {
    // don't tamper with width of Private Use characters
    tattr.attr |= ATTR_WIDE;
}
```

![image](https://user-images.githubusercontent.com/918128/29501393-ad71a852-85fe-11e7-83fc-7d390a1b998d.png)

**Comment by mintty** [comment]
See https://github.com/mintty/mintty/issues/638#issuecomment-284460361,
where I had offered a feature to support wide private use characters (with no feedback so far).
The problem is, however, how to define *which* of the private use characters should be wide and which are narrow. Maybe you can contact the Nerdfonts project whether they have a concept on this.

**Comment by DDuTCH** [comment]
Would it be possible to add a config property where we enter data similar to wide.t?
Then we could enter that in the `.minttyrc` or `config` file.

In the second screenshot I had already added `{ 0xE5FA, 0xF900 },` to wide.t but didn't seem to have any effect.

**Comment by DDuTCH** [comment]
At the least I'd expect the glyphs to start being rendered at x=0 instead of have the left side truncated. 
If there is overhang on the right side I don't mind too much as I'd separate the right side with a space anyway.

Gnome-terminal on CentOS 7
![image](https://user-images.githubusercontent.com/918128/29528380-419d5e5e-8672-11e7-9e4f-5a2e7af52203.png)

**Comment by ryanoasis** [comment]
@DDuTCH Yeah I think the terminal emulator is treating the ambiguous width characters as 'narrow'. In gnome terminal we can change it to 'wide'. Well it's at least an option ¯\_(ツ)_/¯

**Comment by mintty** [comment]
The private use range is in fact marked "Ambiguous width" (just checked; does not apply to non-BMP private use characters).
Mintty has the capability to handle ambiguous-width characters as wide, following the Cygwin concept for that matter, see https://github.com/mintty/mintty/wiki/Tips#ambiguous-width-setting

**Comment by mintty** [comment]
Released 2.8.0.
